### DISCUSSION TITLE
What is LKG?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/aspnetcore#57190] [Q&A] [answered]
`TokenValidationParameters` class has a proprty `ValidateWithLKG`. What is it for? I could not find what this LKG stands for.
[The documentation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.tokens.tokenvalidationparameters.validatewithlkg?view=msal-web-dotnet-latest#microsoft-identitymodel-tokens-tokenvalidationparameters-validatewithlkg) is pretty much useless. =(

**Comment by martincostello** [comment]
My guess would be that it stands for _Last Known Good_.

**Comment by pinkfloydx33** [comment, marked as answer]
It's "Last Known Good", per the [changelog of Microsoft.IdentityModel](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d2632a77aa627707d3e2f04248daea62ab45fd47/CHANGELOG.md?plain=1#L131)

> Introduced an injection point for external metadata management and adjusted the issuer Last Known Good (LKG) to maintain the state within the issuer validator.
